flax/linen/summary.py

```diff
@@ -0,0 +1,431 @@
+
+def _render_table(table: Table) -> str:
+  rich_table = rich.table.Table(
+      show_header=True,
+      show_lines=True,
+      show_footer=True,
+      title=f"{table.module.__class__.__name__} Summary",
+      # box=rich.box.HORIZONTALS,
+  )
+
+  rich_table.add_column("path")
+  # rich_table.add_column("variables")
+  rich_table.add_column("outputs")
+
+  for col in table.collections:
+    rich_table.add_column(col)
+
+  for row in table:
+    collections_size_repr = []
+
+    for collection, size_bytes in row.size_and_bytes(table.collections).items():
+      col_repr = ""
+
+      if collection in row.module_variables:
+        col_repr += _as_yaml_str(jax.tree_map(_format_value, row.module_variables[collection]))
+        col_repr += "\n\n"
+      
+      col_repr += f"[bold]{_size_and_bytes_repr(size_bytes)}[/bold]"
+      collections_size_repr.append(col_repr)
+
+    rich_table.add_row(
+      "/".join(row.path) if row.path else table.module.__class__.__name__,
+      # _as_yaml_str(jax.tree_map(_format_value, row.module_variables)),
+      _as_yaml_str(jax.tree_map(_format_value, row.outputs)),
+      *collections_size_repr
+    )
+
+
+  # add footer with totals
+  rich_table.columns[1].footer = rich.text.Text.from_markup("Total", justify="right")
+
+  # get collection totals
+  colletion_total = {col: (0, 0) for col in table.collections}
+  for row in table:
+    for col, size_bytes in row.size_and_bytes(table.collections).items():
+      colletion_total[col] = (
+        colletion_total[col][0] + size_bytes[0],
+        colletion_total[col][1] + size_bytes[1],
+      )
+
+  # add totals to footer
+  for i, col in enumerate(table.collections):
+    rich_table.columns[2 + i].footer = _size_and_bytes_repr(colletion_total[col])
+
+  # add final totals to caption
+  caption_totals = (0, 0)
+  for (size, bytes) in colletion_total.values():
+    caption_totals = (
+      caption_totals[0] + size,
+      caption_totals[1] + bytes,
+    )
+
+  rich_table.caption_style = "bold"
+  rich_table.caption = f"\nTotal Parameters: {_size_and_bytes_repr(caption_totals)}"
+
+  return "\n" + _get_rich_repr(rich_table) + "\n"
+
+  
+
+def _size_and_bytes_repr(size_and_bytes: Tuple[int, int]) -> str:
+  size, bytes = size_and_bytes
+
+  if size == 0:
+    return ""
+
+  bytes_repr = _format_bytes(bytes)
+
+  return f"{size:,} [dim]({bytes_repr})[/dim]"
+
+def _size_and_bytes(pytree: Any) -> Tuple[int, int]:
+  size = sum(
+    (array.size
+    for array in jax.tree_leaves(pytree)),
+    0
+  )
+  # number of bytes
+  bytes = sum(
+    (array.size * array.dtype.itemsize
+    for array in jax.tree_leaves(pytree)),
+    0
+  )
+  return size, bytes
+
+def _get_rich_repr(obj):
+  import io
+
+  import rich.console
+
+  f = io.StringIO()
+  console = rich.console.Console(file=f, force_terminal=True)
+  console.print(obj)
+
+  return f.getvalue()
+
+def _as_yaml_str(value) -> str:
+  
+  if hasattr(value, "__len__") and len(value) == 0:
```

Comment:
Isn't this more general?

```suggestion
  if not value:
```

Reply:
If `value` is something like `0` or `None` it would give the wrong output.

Reply:
Make sense! Let's keep it as is then.